### PR TITLE
Fix VC support for notification recipients

### PR DESCRIPTION
**Pull request by ViacheslavKlimov**
## Pull Request description

Put your PR description here instead of this sentence.   

## General checklist

- [ ] You have reviewed the guidelines [document](https://docs.google.com/document/d/1wqcOafLx5hth8SAg4dqV_LV3un3m5WYR8RdTJ4MbbUM/edit?usp=sharing).
- [ ] [Labels](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels#about-labels) that classify your pull request have been added.
- [ ] The [milestone](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/about-milestones) is specified and corresponds to fix version.  
- [ ] Description references specific [issue](https://github.com/thingsboard/thingsboard/issues).
- [ ] Description contains human-readable scope of changes.
- [ ] Description contains brief notes about what needs to be added to the documentation.
- [ ] No merge conflicts, commented blocks of code, code formatting issues.
- [ ] Changes are backward compatible or upgrade script is provided.
- [ ] Similar PR is opened for PE version to simplify merge. Crosslinks between PRs added. Required for internal contributors only.
  
## Front-End feature checklist

- [ ] Screenshots with affected component(s) are added. The best option is to provide 2 screens: before and after changes;
- [ ] If you change the widget or other API, ensure it is backward-compatible or upgrade script is present.
- [ ] Ensure new API is documented [here](https://github.com/thingsboard/thingsboard-ui-help)

## Back-End feature checklist

- [ ] Added corresponding unit and/or integration test(s). Provide written explanation in the PR description if you have failed to add tests.
- [ ] If new dependency was added: the dependency tree is checked for conflicts.
- [ ] If new service was added: the service is marked with corresponding @TbCoreComponent, @TbRuleEngineComponent, @TbTransportComponent, etc.
- [ ] If new REST API was added: the RestClient.java was updated, issue for [Python REST client](https://github.com/thingsboard/thingsboard-python-rest-client) is created.
- [ ] If new yml property was added: make sure a description is added (above or near the property).

### CODE REVIEW

File: application/src/main/java/org/thingsboard/server/service/notification/DefaultNotificationSchedulerService.java [hunk 1]
```diff
@@ -48,20 +48,21 @@
...
 @RequiredArgsConstructor
 @Slf4j
-@SuppressWarnings("UnstableApiUsage")
 public class DefaultNotificationSchedulerService extends AbstractPartitionBasedService<NotificationRequestId> implements NotificationSchedulerService {
 
     private final NotificationCenter notificationCenter;
     private final NotificationRequestService notificationRequestService;
-    private final SchedulerComponent scheduler;
     private final NotificationExecutorService notificationExecutor;
+    private final ScheduledExecutorService scheduler = Executors.newSingleThreadScheduledExecutor(ThingsBoardThreadFactory.forName("notification-scheduler"));
```
[the remark this answers is not in the document]
Comment (ashvayka) [reply application/src/main/java/org/thingsboard/server/service/notification/DefaultNotificationSchedulerService.java:65]: :+1:

File: application/src/main/java/org/thingsboard/server/service/sync/ie/exporting/impl/NotificationTargetExportService.java [hunk 2]
```diff
@@ -43,6 +43,9 @@ protected void setRelatedEntities(EntitiesExportCtx<?> ctx, NotificationTarget n
                     CustomerUsersFilter customerUsersFilter = (CustomerUsersFilter) usersFilter;
                     customerUsersFilter.setCustomerId(getExternalIdOrElseInternal(ctx, new CustomerId(customerUsersFilter.getCustomerId())).getId());
                     break;
+                case USER_LIST:
+                    // users list stays as is and is replaced with current user id on import (due to user entities not being supported by VC)
+                    break;
```
[the remark this answers is not in the document]
Comment (ViacheslavKlimov) [reply application/src/main/java/org/thingsboard/server/service/sync/ie/exporting/impl/NotificationTargetExportService.java:48]: ?

File: application/src/main/java/org/thingsboard/server/service/sync/ie/importing/impl/NotificationTargetImportService.java [hunk 3]
```diff
@@ -66,10 +66,7 @@ protected NotificationTarget prepare(EntitiesImportCtx ctx, NotificationTarget n
                     break;
                 case USER_LIST:
                     UserListFilter userListFilter = (UserListFilter) usersFilter;
-                    userListFilter.setUsersIds(userListFilter.getUsersIds().stream()
-                            .map(UserId::new).map(idProvider::getInternalId)
-                            .map(UUIDBased::getId).collect(Collectors.toList())
-                    );
+                    userListFilter.setUsersIds(List.of(ctx.getUser().getUuidId())); // user entities are not supported by VC; replacing with current user id
```
[the remark this answers is not in the document]
Comment (ViacheslavKlimov) [reply application/src/main/java/org/thingsboard/server/service/sync/ie/importing/impl/NotificationTargetImportService.java:69]: What do you suggest to do in the `default` section?

File: application/src/main/java/org/thingsboard/server/service/sync/ie/exporting/impl/NotificationTargetExportService.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (smatvienko-tb) [reply application/src/main/java/org/thingsboard/server/service/sync/ie/exporting/impl/NotificationTargetExportService.java:48]: @ViacheslavKlimov do some default action or throw an exception that actually never fires until some new code changes appear. It's up to your business logic. Just have a default section

**Pull request merged**